QualityControl/lib/services/QcdbDownload.service.js

```
@@ -0,0 +1,107 @@
+      const filename = contentDisposition?.slice(17, contentDisposition.length - 1);
+      this._logger.infoMessage(`ROOT size: ${contentLength}`);
+      // We will stream the data from QCDB's answer directly back to the user.
+      if (res != undefined) {
+        res.setHeader('Content-Type', contentType ?? 'application/root');
+        res.setHeader('Content-Disposition', contentDisposition ?? `attachment; filename="${filename}"`);
+        if (contentLength) {
+          res.setHeader('Content-Length', contentLength);
+        }
+
+        await this._pipelineAsync(response.body, res);
+        return true;
+      } else {
+        // We'll return the file from the response back.
+        const blob = await response.blob();
+        const file = new File([blob], filename ?? 'export.root', { type: contentType ?? 'application/root' });
+        return file;
+      }
+    } catch (error) {
+      this._logger.errorMessage(error);
+      return false;
+    }
+  }
+
+  /**
+   * Get qcdb root objects from qcdb.
+   * Only a single root object request supported for now.
+   * @param {string} objectIds - Id of ROOT object to request.
+   * @param {Express.Response} res - Express response object.
+   */
+  async getQcdbRootObjects(objectIds, res) {
+    const promises = [];
+    if (typeof objectIds === 'string') {
+      promises.push(this.requestObject(objectIds, res));
+    } else {
+      // Technically this can be stopped at the DTO layer but multiple id's will be implemented soon.
+      res.status(500).send('Option to retrieve more than 1 ROOT object not implemented yet.');
+    }
+
+    try {
+      const files = await Promise.all(promises);
+      if (files.filter((file) => file === false).length > 0) {
+        throw new Error('qcdb object request failed.');
+      }
+    } catch (error) {
+      this._logger.errorMessage(error.message);
+      res.status(500).send("Unable to retrieve ROOT object('s)");
```

The service should not deal with using the response object to send an error. 
This method should instead let the error bubble up to the controller which needs to catch it
Same for the error on L94